compiler/rustc_mir_transform/src/uninhabited_enum_branching.rs

```diff
@@ -103,20 +104,29 @@ impl<'tcx> MirPass<'tcx> for UninhabitedEnumBranching {
             let terminator = bb_data.terminator();
             let TerminatorKind::SwitchInt { targets, .. } = &terminator.kind else { bug!() };
 
-            let mut reachable_count = 0;
             for (index, (val, _)) in targets.iter().enumerate() {
-                if allowed_variants.contains(&val) {
-                    reachable_count += 1;
-                } else {
+                if !allowed_variants.remove(&val) {
                     removable_switchs.push((bb, index));
                 }
             }
 
-            if reachable_count == allowed_variants.len() {
+            if allowed_variants.is_empty() {
                 removable_switchs.push((bb, targets.iter().count()));
+            } else if allowed_variants.len() == 1 {
+                #[allow(rustc::potential_query_instability)]
```

Comment:
Try using an IndexSet instead of picking a random item. If perf is happy with it, too, we'll go that route.

Reply:
There is a single item, so that's not random. Still deserves a comment to justify the attribute.

Reply:
🤦 yea was so focussed on the creation of the set, I forgot that there is this one valid use for it 😆

Reply:
I can convert to `Vec`, but I'm not sure if that makes sense.

Reply:
nah, this was just me not getting it. a comment is totally sufficient